### PR TITLE
Other/reconnection support

### PR DESCRIPTION
**Pull request by NikolaiShipilov**

### CODE REVIEW

File: src/session.cpp [hunk 1]
```diff
@@ -39,7 +44,7 @@ void Session::setOptions()
 
 void Session::close(OnCompleteCallback onClosedCallback)
 {
-    NS_LOG_T("Closing {} session", (role == boost::beast::role_type::server) ? "server" : "client");
+    NS_LOG_I("Closing {} session", (role == boost::beast::role_type::server) ? "server" : "client");
```
Comment (JakaMohorkoDS) [inline src/session.cpp:47]: In general, I'd avoid too many Info logs. I'd move this to debug.

File: src/session.cpp [hunk 2]
```diff
@@ -50,22 +55,41 @@ void Session::close(OnCompleteCallback onClosedCallback)
...
                                   if (wsStream->is_open())
                                   {
-                                      NS_LOG_W("Closing {} session failure: {}", roleName, ec.message());
+                                      NS_LOG_E("Closing {} session failure: {}", roleName, ec.message());
                                       onClosedCallback(ec);
                                   }
                                   else
                                   {
-                                      NS_LOG_T("Closed {} session", roleName);
+                                      NS_LOG_I("{} session successfully closed", roleName);
```
Comment (JakaMohorkoDS) [inline src/session.cpp:63]: Similarly here, I'd probably keep session information as debug, but have a more understandable message logged as info in the SDK (eg. device successfully disconnected).

File: changelog [hunk 3]
```diff
@@ -1,2 +1,9 @@
 ## v1.0.0
  - Initial tag
+
+## v1.0.1
+ - Add different callbacks for Client resolve/connect/handshake errors
+ - Add support for IPv6 connection
+ - Increase size of websocket write buffer
+ - Add heartbeat mechanism based on websocket ping/pong datagrams
```
Comment (JakaMohorkoDS) [inline changelog:8]: Are heartbeat timeouts handled here, or should it be handled by users of the this library?

**Review by JakaMohorkoDS — APPROVED** [review]

File: changelog (hunk 3 above)
Comment (NikolaiShipilov) [reply changelog:8]: Library users specify the duration of the heartbeat period, and the mechanism for detecting disconnections due to heartbeat timeout expiration is implemented external to the library.

**Pull request merged**